source/features/tag-changelog-link.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+Adds a compare link on each releases/tags/single tag page so that you can see what has changed since the previous release.
+If the tags are namespaced then it tries to get the previous release of the same namespaced tag.
+
+See it in action at: https://github.com/parcel-bundler/parcel/releases
+*/
+import './tag-changelog-link.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import fetchDom from '../libs/fetch-dom';
+import {diff} from '../libs/icons';
+import {isSingleTagPage} from '../libs/page-detect';
+import {getRepoPath, getRepoURL} from '../libs/utils';
+
+type TagDetails = {
+	element: HTMLElement;
+	commit: string;
+	tag: string;
+}
+
+async function getNextPage(): Promise<DocumentFragment> {
+	const nextPageLink = select<HTMLAnchorElement>('.pagination a:last-child');
+	if (nextPageLink) {
+		return fetchDom(nextPageLink.href);
+	}
+
+	if (isSingleTagPage()) {
+		const [, tag = ''] = getRepoPath()!.split('releases/tag/', 2); // Already URL-encoded
+		return fetchDom(`/${getRepoURL()}/tags?after=${tag}`);
+	}
+
+	return new DocumentFragment();
+}
+
+async function init(): Promise<void | false> {
+	if (select.exists('.blankslate')) {
+		return false;
+	}
+
+	const tagRegex = /\/releases\/tag\/(.*)/;
+	const documents = [document, await getNextPage()] as any; // TODO: fix select-dom types to accept mixed arrays
+
+	// not(.js-timeline-tags-expander) is needed as there can be some collapsed tags
+	// See https://github.com/facebook/react/releases?after=v16.7.0 for an example
+	const tagContainerSelector = '.release, .Box-row .commit, .release-entry .release-main-section:not(.commit):not(.js-timeline-tags-expander)';
+
+	// These selectors need to work on:
+	// https://github.com/facebook/react/tags (tags list)
+	// https://github.com/facebook/react/releases (releases list)
+	// https://github.com/parcel-bundler/parcel/releases (releases list without release notes)
+	const tagSelector = 'h4.commit-title > a[href*="/releases/tag/"], div.release-header .f1 > a[href*="/releases/tag/"]';
+
+	const allTags: TagDetails[] = select.all(tagContainerSelector, documents).map(element => ({
+		element,
+		tag: decodeURIComponent(select<HTMLAnchorElement>(tagSelector, element)!.pathname.match(tagRegex)![1]),
```

Comment:
I think this isn't needed, also because the tag will end up back into a URL anyway
```suggestion
		tag: select<HTMLAnchorElement>(tagSelector, element)!.pathname.match(tagRegex)![1],
```

Reply:
We are also showing the tooltip so if we don't `decodeURIComponent` it will display tooltip text in the encoded form.

Reply:
Indeed. Then maybe we need to move `decodeURIComponent` to the tooltip because we need the tag to be urlencoded in the URL